### PR TITLE
[driver] Add option for measurement resolution to DS18B20 driver + fix MacOS CI

### PR DESCRIPTION
**Pull request by chris-durand**
Add option to configure measurement resolution to DS18B20 temperature sensor driver. Tested in hardware.

### CODE REVIEW

File: src/modm/driver/temperature/ds18b20.hpp [hunk 1]
```diff
@@ -17,15 +18,25 @@
 
 namespace modm
 {
+	struct ds18b20
```
Comment (salkinium) [inline src/modm/driver/temperature/ds18b20.hpp:32]:
```suggestion
	/// \ingroup	modm_driver_ds18b20
	struct ds18b20
```

**Review by salkinium — COMMENTED** [review]
Nice!

File: src/modm/driver/temperature/ds18b20.hpp (hunk 1 above)
Comment (chris-durand) [reply src/modm/driver/temperature/ds18b20.hpp:32]: Fixed. I have also added documentation for the enum and added the `configure()` method to the existing example that it gets instantiated when compiling the driver in the CI.

**Review by salkinium — APPROVED** [review]
Thanks!

**Comment by chris-durand** [comment]
@salkinium The make build of Mac OS hosted examples is still broken:
```
Building: linux/build_info/main.cpp with Make
Compiling C···· release/modm/ext/printf/printf.o/bin/sh: gcc-10: command not found
make: *** [build/make-release/modm/ext/printf/printf.o] Error 127
```

The make generator has gcc-10 hard-coded for Mac OS hosted, but the pipeline only has 11:
```
%% if platform == "hosted"
        UNAME_S := $(shell uname -s)
        ifeq ($(UNAME_S),Darwin)
                C_SUFFIX := -10
        # Using homebrew include and lib paths on macOS
        CPPDEFINES += -I/usr/local/include
        LIBPATH += -L/usr/local/lib
        endif
%% endif
```

Can I change this to 11 or do you want to fix this properly?

**Comment by salkinium** [comment]
> Can I change this to 11 or do you want to fix this properly?

You can change it to 11, very few people if any use modm on macOS with make anyways.

**Pull request merged**